app/controllers/api/groups_controller.rb

```diff
@@ -369,6 +369,35 @@ def extension
       render 'shared/http_status', locals: { code: '422', message: e.to_s }, status: :unprocessable_entity
     end
 
+    def collect_submission
+      @grouping = Grouping.find_by(group_id: params[:id], assignment: params[:assignment_id])
+      unless @grouping.current_submission_used.nil?
+        released = @grouping.current_submission_used.results.exists?(released_to_students: true)
+        if released
+          render 'shared/http_status', locals: { code: '422', message:
+            HttpStatusHelper::ERROR_CODE['message']['422'] }, status: :unprocessable_entity
+          return
+        end
+      end
+      apply_late_penalty = if params[:apply_late_penalty].nil? || params[:apply_late_penalty] == false
```

Comment:
You don't need the `|| params[:apply_late_penalty] == false`, it's handled by the `else` case. Similar comment to the one below